### PR TITLE
Drukte

### PR DESCRIPTION
**Pull request by rikvanderwerf**
https://tree.taiga.io/project/melloww-smart-gym/us/9?kanban-status=718746

### CODE REVIEW

File: alembic/versions/51578b5cfa93_user_activity_and_busyness.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    ### end Alembic commands ###
+
+
+def downgrade():
+    ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('sport_schedule',
+    sa.Column('id', postgresql.UUID(), autoincrement=False, nullable=False),
+    sa.Column('user_id', postgresql.UUID(), autoincrement=False, nullable=True),
+    sa.Column('name', sa.VARCHAR(length=100), autoincrement=False, nullable=True),
+    sa.Column('reminder_minutes', sa.INTEGER(), autoincrement=False, nullable=True),
```
Comment (robinsiep) [inline alembic/versions/51578b5cfa93_user_activity_and_busyness.py:39]: you delete more than you create

File: smartgymapi/handlers/busyness.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+from smartgymapi.lib.factories.busyness import BusynessFactory
+from smartgymapi.lib.validation.busyness import BusynessSchema
+from smartgymapi.lib.validation.user_activity import UserActivitySchema
+
+log = logging.getLogger(__name__)
+
+
+@view_defaults(containment=BusynessFactory,
+               permission='public',
```
Comment (robinsiep) [inline smartgymapi/handlers/busyness.py:18]: permissions? doubt  this should be public

File: smartgymapi/handlers/busyness.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        return self.hour_count
+
+    def fill_hour_count(self, activities, predict_for_today=False,
+                        predict=False):
+        if predict:
+            amount_of_days = 0
+            for (year, items) in groupby(activities, grouper):
+                amount_of_days += 1
+                for item in items:
+                    if (predict_for_today and
```
Comment (robinsiep) [inline smartgymapi/handlers/busyness.py:75]: Might wanna add some comments.

File: smartgymapi/handlers/busyness.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                self.hour_count[hour] = round(
+                    self.hour_count[hour] / amount_of_days)
+        else:
+            for item in activities:
+                self.add_item_to_hour_count(item)
+
+    def add_item_to_hour_count(self, item):
+        self.hour_count[str(item.start_date.hour)] = \
+            self.hour_count.setdefault(
+            str(item.start_date.hour), 0) + 1
```
Comment (robinsiep) [inline smartgymapi/handlers/busyness.py:92]: comments

File: smartgymapi/handlers/busyness.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def grouper(item):
+    return item.start_date.year
```
Comment (robinsiep) [inline smartgymapi/handlers/busyness.py:96]: docstring

File: smartgymapi/lib/factories/root.py [hunk 6]
```diff
@@ -1,13 +1,18 @@
 from smartgymapi.lib.factories.user import UserFactory
+from smartgymapi.lib.factories.user_activity import UserActivityFactory
+from smartgymapi.lib.factories.busyness import BusynessFactory
```
Comment (robinsiep) [inline smartgymapi/lib/factories/root.py:3]: import ordering

File: smartgymapi/lib/factories/root.py [hunk 7]
```diff
@@ -1,13 +1,18 @@
...
+
     def __init__(self, request):
         self.requires_oauth = False
         self.request = request
         self.__name__ = None
         self.__parent__ = None
 
         self['user'] = UserFactory(self, 'user')
+        self['user_activity'] = UserActivityFactory(self, 'user_activity')
+        self['busyness'] = BusynessFactory(self, 'busyness')
```
Comment (robinsiep) [inline smartgymapi/lib/factories/root.py:17]: ordering

File: smartgymapi/models/user_activity.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
+import logging
+
+import uuid
+
+from smartgymapi.models.meta import Base, DBSession as session, LineageBase
+
+from sqlalchemy import Column, Date, DateTime, ForeignKey, cast
```
Comment (robinsiep) [inline smartgymapi/models/user_activity.py:7]: import ordering.

File: smartgymapi/models/user_activity.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+def list_user_activities(date=None):
+    q = session.query(UserActivity)
+    if date:
+        q = q.filter(cast(UserActivity.start_date, Date) == date)
+    return q
+
+
+def list_user_activities_for_prediction(date):
+    q = session.query(UserActivity)
+    q = q.filter(
```
Comment (robinsiep) [inline smartgymapi/models/user_activity.py:55]: this method can be one line

File: smartgymapi/models/user_activity.py (hunk 8 above)
Comment (rikvanderwerf) [inline smartgymapi/models/user_activity.py:7]: @Melloww wat dan?

Comment (robinsiep) [inline smartgymapi/models/user_activity.py:7]: > Imports should be grouped in the following order:
> standard library imports
> related third party imports
> local application/library specific imports
> You should put a blank line between each group of imports.

https://www.python.org/dev/peps/pep-0008/#imports

**Pull request closed without merging**